src/Workspaces/Core/Portable/Rename/RenameLocation.ReferenceProcessing.cs

```diff
@@ -309,11 +309,18 @@ private static string TrimNameToAfterLastDot(string name)
                 {
                     if (location.IsInSource)
                     {
-                        Contract.ThrowIfNull(location.SourceTree);
-                        results.Add(new RenameLocation(
-                            location,
-                            solution.GetRequiredDocument(location.SourceTree).Id,
-                            isRenamableAccessor: isRenamableAccessor));
+                        var document = solution.GetRequiredDocument(location.SourceTree);
+
+                        // If the location is in a source generated file, we won't rename it. Our assumption in this case is we
+                        // have cascaded to this symbol from our original source symbol, and the generator will update this file
+                        // based on the renamed symbol.
+                        if (document is not SourceGeneratedDocument)
```

Comment:
i don't love that one has to do a type-check here.  is there no property on Document itself to ask this question?

Reply:
At the moment, no, it's just a type check. We have https://github.com/dotnet/roslyn/issues/50912 tracking to investigate if switching a kind is helpful, although that could also break more than it helps depending.